arb_os/inbox.mini

```diff
@@ -228,174 +150,72 @@ impure func incomingRequestFromRaw(newMsgRaw: IncomingRequestFromInbox, adminMod
-            let rawSideloadMsg = asm(inbox.blockNum,) any { sideload };
-            if (rawSideloadMsg == ()) {
-                return inbox with {
-                    queue: incomingRequestQueue_put(
-                        inbox.queue,
-                        endOfBlockMessage(inbox.blockNum, inbox.timestamp),
-                    )
-                } with {
-                    blockNum: inbox.blockNum+1
-                } with {
-                    seenMsgAtThisBlockNum: false
-                };
-            } else {
-                let newMsg = incomingRequestFromRaw(unsafecast<IncomingRequestFromInbox>(rawSideloadMsg), true);
-                if (newMsg.blockNumber == 0) {
-                    newMsg = newMsg with { blockNumber: inbox.blockNum };
-                }
-                if (newMsg.timestamp == 0) {
-                    newMsg = newMsg with { timestamp: inbox.timestamp };
-                }
-                return inbox with {
-                    queue: incomingRequestQueue_put(inbox.queue, newMsg)
-                };
-            }
+            // Inboxpeek says we hit end of L1 block.
+            // This implies we have hit the end of an Arbitrum block, so do the end of block things.
+            inbox = endOfArbitrumBlock(inbox);
+            return inbox with {
+                ethBlockNum: inbox.ethBlockNum + 1
+            } with {
+                seenMsgAtThisEthBlockNum: false
+            };
         }
     }
 
     let newMsg = incomingRequestFromRaw(asm() IncomingRequestFromInbox { inbox }, false);
-
-    if (newMsg.blockNumber == 0) {
-        newMsg = newMsg with { blockNumber: inbox.blockNum };
+    if (newMsg.arbBlockNumber > inbox.arbBlockNum) {
+        inbox = endOfArbitrumBlock(inbox);
     }
-    if (newMsg.timestamp == 0) {
-        newMsg = newMsg with { timestamp: inbox.timestamp };
-    }
-
 
     return inbox with {
         queue: incomingRequestQueue_put(inbox.queue, newMsg)
     } with {
-        blockNum: newMsg.blockNumber
+        ethBlockNum: newMsg.ethBlockNumber
+    } with {
+        arbBlockNum: newMsg.arbBlockNumber
     } with {
         timestamp: newMsg.timestamp
     } with {
-        seenMsgAtThisBlockNum: true
+        seenMsgAtThisEthBlockNum: true
+    } with {
+        seenMsgAtThisArbBlockNum: true
     };
 }
 
-impure func getFromL1InboxSequencer(inbox: Inbox, seq: SequencerState) -> Inbox {
-    let newMsg = incomingRequestFromRaw(asm() IncomingRequestFromInbox { inbox }, false);
-
-    let blockNum = newMsg.blockNumber;
-    if (blockNum == 0) {
-        blockNum = seq.lastBlockNumSeen;
-    } else {
-        seq = seq with { lastBlockNumSeen: blockNum };
-    }
-    let timestamp = newMsg.timestamp;
-    if (timestamp == 0) {
-        timestamp = seq.lastBlockNumSeen;
-    } else {
-        seq = seq with { lastTimestampSeen: timestamp };
+impure func endOfArbitrumBlock(inbox: Inbox) -> Inbox {
+    let rawSideloadMsg = asm(inbox.arbBlockNum,) any { sideload };
```

Comment:
It seems like this will make the side loaded message run in the next arbitrum block rather than the current one since it increments `arbBlockNum`

Reply:
I don't think so.  The incoming message is assigned its arbBlockNum by `incomingRequestFromRaw` which runs before `inbox.arbBlockNum` gets incremented.

Reply:
Makes sense